packages/wrangler/src/deploy/deploy.ts

```diff
@@ -965,6 +1001,88 @@ function deployWfpUserWorker(
 	logger.log("Current Deployment ID:", deploymentId);
 }
 
+interface SourceMapV3 {
+	version: 3;
+	file: string;
+	sourceRoot: string;
+	sources: string[];
+	sourcesContent: string[];
+	mappings: string;
+	names: string[];
+}
+
+/**
+ * Load and normalize a source map emitted by Wrangler using the given path and
+ * directory metadata.
+ */
+function loadBundledSourceMap(
+	{ name, filePath }: CfModule,
+	sourceMapPath: string,
+	metadata: SourceMapMetadata
+): CfWorkerSourceMap[] {
+	if (filePath === undefined) {
+		return [];
+	}
+	const map = JSON.parse(readFileSync(sourceMapPath).toString()) as SourceMapV3;
+	// Overwrite the file property of the source map to match the name of the
+	// main module in the multipart upload.
+	map.file = name;
+	// Remove the temporary directory prefix generated by Wrangler that appears
+	// in the source root path.
+	map.sourceRoot = map.sourceRoot.slice(metadata.tmpDir.length);
+	map.sources = map.sources.map((source) => {
+		const originalPath = path.join(path.dirname(filePath), source);
+		// Remove the temporary build directory prefix generated by Wrangler
+		// that appears in the source path.
+		return originalPath.slice(metadata.entryDirectory.length + 1);
+	});
+	return [
+		{
+			name: name + ".map",
+			content: JSON.stringify(map),
+		},
+	];
+}
+
+/**
+ * Find source maps by scanning module contents for special `//#
+ * sourceMappingURL=` comments pointing to source map files.
+ */
+function scanSourceMaps(modules: CfModule[]): CfWorkerSourceMap[] {
+	const maps: CfWorkerSourceMap[] = [];
+	for (const module of modules) {
+		const commentPrefix = "//# sourceMappingURL=";
+		const commentIndex = module.content.indexOf(commentPrefix);
```

Comment:
nit: we should probably just check last line for correctness, probably wont actually impact anyone but ya-know.